jekyll/_cci2/circleci-images.md

```diff
@@ -51,34 +52,34 @@ If you need a generic image to run on CircleCI, to use with orbs, or to use as a
 ```
 
-This is a direct replacement for the legacy CircleCI Go image (`circleci/golang`). It brings with it better documentation, more determinism, and benefits from the highly efficient infrastructure it’s built on.
+This is a direct replacement for the legacy CircleCI Go image (`circleci/golang`). Note, the Docker Hub repository is `cimg`.
 
 
 ## Best Practices
 
-The convenience images in the following sections are based on the most recently built versions of upstream images, so it is best practice to use the most specific image possible. This makes your builds more deterministic by preventing an upstream image from introducing unintended changes to your image.
-
-CircleCI bases pre-built images off of upstream, for example, `circleci/ruby:2.4-node` is based off the most up to date version of the Ruby 2.4-node container. Using `circleci/ruby:2.4-node` is similar to using `:latest`. It is best practice to lock down aspects of your build container by specifying an additional tag to pin down the image in your configuration.
+The next-gen convenience images in the following sections are based on the most recent LTS Ubuntu docker images and installed with the base libraries for the language or services, so it is best practice to use the most specific image possible. This makes your builds more deterministic by preventing an upstream image from introducing unintended changes to your image. The legacy images are Debian `jessie` or `stretch`.
 
-That is, to prevent unintended changes that come from upstream, instead of using `circleci/ruby:2.4-node` use a more specific version of these containers to ensure the image does not change with upstream changes until you change the tag.
+That is, to prevent unintended changes that come from upstream, instead of using `cimg/ruby:2.4-node` use a more specific version of these containers to ensure the image does not change with upstream changes until you change the tag.
 
-For example, add `-jessie` or `-stretch` to the end of each of those containers to ensure you’re only using that version of the Debian base OS. Pin down those images to a specific point version, like `circleci/ruby:2.3.7-jessie`, or specify the OS version with `circleci/ruby:2.3-jessie`. Specifying the version is possible for any of the CircleCI images.
+For example, pin down those images to a specific point version, like `cimg/ruby:2.4.10-node`. Specifying the version is possible for any of the CircleCI images.
 
-It is also possible to specify all the way down to the specific SHA of the image you want to use. Doing so allows you to test specific images for as long as you like before making any changes.
+It is also possible to specify all the way down to the specific SHA of the image you want to use. For example, you can use `cimg/ruby@sha256:e4aa60a0a47363eca3bbbb066620f0a5967370f6469f6831ad52231c87ca9390` instead of `cimg/ruby:2.4.10-node`. Doing so allows you to test specific images for as long as you like before making any changes.
```

Comment:
This isn't recommended by us though. If there's a security or major bug, we can't fix the image for them. I'd maybe add this as a note.

Reply:
Was reading through the docs and found this. I apologize for writing on this ancient PR discussion, but I was looking where the text came from.

What I found interesting on the docs is the recommendations to avoid pinning an image and instead using a tag.

As the docs are now, it is a bit confusing IMO. While the reason of having security updates on the build is a valid one, the note should also say that it has the potential to make the builds not immutable and creating unexpected results.

Another aspect is a security one, it might not happen, but if any image on the cimg namespace gets compromised, a simple push to a new tag exposes all people using those images, while having it pinned would not have that issue.

Let me know what you think about it, and If you like I can create a PR, to start a discussion there instead of this old PR.

Again, sorry for intruding.